src/Billing/AutoBilling.php

```diff
@@ -0,0 +1,327 @@
+        }
+    }
+
+    public function generateBilling($dx, $cpt)
+    {
+        $event_date = date("Y-m-d H:i:s");
+
+        /**
+         *  Aggregate information to fill billing table
+         */
+        $billingData = explode("-", $cpt);  //separate up billing information into code and text
+        $code = trim($billingData[0]);  // code that is in the calendar description with the decimalOpen
+        $text = $billingData[1]; //get the billing description from the code table
+
+
+        $icd10code = explode(" - ", $dx);
+        $icd10 = $icd10code[0];
+        $desc = $icd10code[1];
+
+        /**
+         * Grab Fee
+         * @returns int
+        */
+        $fees = $this->getCodeFee($code);
+
+        /**
+         * Grab the encounter to post charges using the event date
+         * This file has been tailored to other areas submitting data
+         * to create billing. I left them in case they are needed in the future
+         * like doing billing from the calendar
+         * @returns int
+         */
+       //$enc = $this->getEventEncounter($pid, $event_date);
+        $enc = $_SESSION['encounter'];
+        /**
+         * Grab the provider for the encounter from the event table
+         */
+        //$provider = $this->getEventProvider($pid, $event_date);
+        $provider = $_SESSION['authUserID'];
+        /**
+         * find out if there are any entries in the billing table for event date for this patient
+         * moved to calendar inc
+         *
+         */
+        $hasBilling = $this->findEventBilling($this->pid, $enc);
+
+        /**
+         * Gets the modifier assigned to the
+         * If there are modifiers in the code table
+         */
+        $modifier = $this->getModifier($code);
+        $modifier = str_replace(",", "", $modifier);
+
+        //insert billing if there is note and entry for this particular encounter
+        if (empty($hasBilling['encounter'])) {
+            if (!empty($code)) {
+                    $codetype = "CPT4";
+                    $this->insertCPTBilling($event_date, $code, $this->pid, $provider, $this->userid, $enc, $text, $fees, $icd10, $codetype, $modifier);
+            }
+
+            //Enter ICD10 if it exist
+            if (!empty($desc) && empty($res)) {
+                    $codetype = "ICD10";
+                    $this->insertICDBilling($event_date, $codetype, $icd10, $this->pid, $provider, $this->userid, $enc, $desc);
+            }
+        }
+    }// End of billing entries
+
+    /**
+     * @param $event_date
+     * @param $code
+     * @param $pid
+     * @param $provider
+     * @param $userid
+     * @param $enc
+     * @param $text
+     * @param $fees
+     * @param $icd
+     * @param $codetype
+     * @param $modifier
+     */
+    private function insertCPTBilling($event_date, $code, $pid, $provider, $userid, $enc, $text, $fees, $icd10, $codetype, $modifier)
+    {
+        $sql = "REPLACE INTO billing SET " .
```

Comment:
REPLACE is a MySQL extension to the SQL standard. So for a project that avoids universal things like SQL views and SQL functions, it is best to implement Add/Change logic rather than this practice of clobbering existing record when the function is clearly called insertXXX.

This will apply to all REPLACE statements.

Reply:
"Add/Change logic" enlighten me, please. Show me Add/Change logic in the system, please.

Reply:
I think he means insert/update.